projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -203,6 +203,17 @@ public function init_hook_action() {
 		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
 		add_filter( 'jetpack_sync_post_meta_whitelist', array( $this, 'allow_sync_post_meta' ) );
 		$this->setup_cpts();
+
+		if ( Jetpack::is_module_active( 'subscriptions' ) && jetpack_is_frontend() ) {
```

Comment:
Instead of using `jetpack_is_frontend`, why not using `( new Host() )->is_wpcom_simple() )` which seems more explanatory.

Reply:
Not sure what you mean by that. I wanted to prevent adding the `subscriber_logout` action on API requests, etc.  and it seems `jetpack_is_frontend` is doing it.

We need it on all site types, not only on Simple.

Reply:
On self-hosted (and I think onWoA) it will not work as expected. We login on WPCOM but the cookie is not set there.